pkg/vsphere/disk/disk_manager.go

```diff
@@ -421,3 +423,51 @@ func (m *Manager) devicePathByURI(op trace.Operation, datastoreURI *object.Datas
 
 	return sysPath, nil
 }
+
+// AttachAndMount creates and attaches a vmdk as a non-persistent disk, mounts it, and returns the mount path.
+func (m *Manager) AttachAndMount(op trace.Operation, datastoreURI *object.DatastorePath) (string, error) {
+	config := NewNonPersistentDisk(datastoreURI)
+	d, err := m.CreateAndAttach(op, config)
```

Comment:
CreateAndAttach either creates and attaches a disk or just attaches a disk so we are all good here. That one function doing multiple things using magic was what I didn't like and hoped to change.